pkg/controller/deployment/operator_log_level.go

```diff
@@ -0,0 +1,37 @@
+package deployment
+
+import (
+	"strconv"
+
+	"k8s.io/component-base/logs"
+	"k8s.io/klog/v2"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+var operatorLogLevels = map[operatorv1.LogLevel]int{
+	operatorv1.Normal:   2,
+	operatorv1.Debug:    4,
+	operatorv1.Trace:    6,
+	operatorv1.TraceAll: 8,
+}
+
+// setOperatorLogLevel is used to set the value of klog verbosity
+// if current verbosity level does not match logLevel
+func setOperatorLogLevel(logLevel operatorv1.LogLevel) error {
+	if logLevel == "" {
+		logLevel = operatorv1.Normal
+	}
+
+	v := klog.Level(operatorLogLevels[logLevel])
+	// set verbosity by checking if V(n) and V(n+1) are enabled,
```

Comment:
Can you add a comment on why `V(n)` and `V(n+1)` are considered as well?

Reply:
Yeah, I can do that; context being there isn't a direct straight-forward way to check the current verbosity level at runtime (atleast that I know of going through `klog` and `k8s.io/component-base/logs`) which brought me up to this logic to compare with the help of V(n) and V(n+1) enabled or not at runtime.

Reply:
https://github.com/openshift/cert-manager-operator/pull/111#discussion_r1169705175